ibm_satellite_attach_host_script shows incorrect syntax for labels

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->
### New or Affected Resource(s) or Datasource(s)

* ibm_satellite_attach_host_script

### Description

ibm_satellite_attach_host_script documentation is incorrect: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/data-sources/satellite_attach_host_script. It shows labels as being of the form `cpu=4`. However, they must be of the form `cpu:4`, otherwise this causes a crash in the Terraform provider:

```
tack trace from the terraform-provider-ibm_v1.46.0 plugin:

panic: runtime error: index out of range [1] with length 1

goroutine 771 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex.FlattenHostLabels(0x14002ed9670, 0x1, 0x1, 0x1)
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex/structures.go:2745 +0x158
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/satellite.dataSourceIBMSatelliteAttachHostScriptRead(0x14002c99a80, 0x103a6cae0, 0x140004d5b00, 0x103a849e0, 0x0)
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/satellite/data_source_ibm_satellite_host_script.go:123 +0x414
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1400094f5e0, 0x103b141a0, 0x14002c2a300, 0x14002c99a80, 0x103a6cae0, 0x140004d5b00, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:712 +0x170
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0x1400094f5e0, 0x103b141a0, 0x14002c2a300, 0x14002c99980, 0x103a6cae0, 0x140004d5b00, 0x140004d5b00, 0x14002c99980, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:943 +0x118
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0x1400000d578, 0x103b140f8, 0x14002c2a300, 0x14003da3220, 0x102c8cb3c, 0x12, 0x0)
```

I confirmed this by inspecting the provider code.

--- a/ibm/flex/structures.go
+++ b/ibm/flex/structures.go
@@ -2741,9 +2741,10 @@ func FlattenHostLabels(hostLabels []interface{}) map[string]string {
 	labels := make(map[string]string)
 	for _, v := range hostLabels {
 		parts := strings.Split(v.(string), ":")
-		if parts != nil {
-			labels[parts[0]] = parts[1]
+		if len(parts) != 2 {
+			log.Fatal("Entered label " + v.(string) + "is in incorrect format.")
 		}
+		labels[parts[0]] = parts[1]
 	}
 
 	return labels

--- a/website/docs/d/satellite_attach_host_script.html.markdown
+++ b/website/docs/d/satellite_attach_host_script.html.markdown
@@ -16,7 +16,7 @@ Retrieve information of an existing IBM Satellite location registration script a
 ```terraform
 data "ibm_satellite_attach_host_script" "script" {
   location          = var.location
-  labels            = var.labels
+  labels            = ["cpu:4"]
   host_provider     = "ibm"
 }
 ```
@@ -55,7 +55,7 @@ Review the argument references that you can specify for your data source.
 - `custom_script` - (Optional, String) RHEL hosts only. The custom script that has to be appended to generated host script file. Either `custom_script` or `host_provider` is required. This `custom_script` will be appended to the downloaded host attach script. Find custom scripts for respective cloud providers [aws](https://cloud.ibm.com/docs/satellite?topic=satellite-aws#aws-host-attach), [google](https://cloud.ibm.com/docs/satellite?topic=satellite-gcp#gcp-host-attach), [azure](https://cloud.ibm.com/docs/satellite?topic=satellite-azure#azure-host-attach), [ibm](https://cloud.ibm.com/docs/satellite?topic=satellite-ibm#ibm-host-attach).
 - `location` - (Required, String) The name or ID of the Satellite location.
 - `host_provider` - (Optional, String) The name of host provider, such as `ibm`, `aws` or `azure`.
-- `labels` - (Optional, Strings) The key-value pairs to label the host, such as `cpu=4` to describe the host capabilities.
+- `labels` - (Optional, Set(Strings)) The set of key-value pairs to label the host, such as `["cpu:4"]` to describe the host capabilities.
 - `script_dir` - (Optional, String) The directory path to store the generated script.
 
 ## Attributes reference

